### PR TITLE
Temporal signal split fix

### PR DESCRIPTION
**Pull request by tforgaard**
Hi,

This fix ensures that additional features are properly split between the train and test dataset when using `temporal_signal_split`.

I edited a couple of the tests that use `temporal_signal_split`. Do leave feedback if something needs fixing.

Thanks

### CODE REVIEW

**Comment by benedekrozemberczki** [comment]
Can you pull and increase the test coverage with examples?

**Comment by tforgaard** [comment]
I am a bit at a loss as to writing examples. 

However, I noticed that two of the tests in `dataset_test` were named `test_discrete_train_test_split_dynamic` which I think might cause one of them to not be run. I renamed one of them now
